openassessment/xblock/staff_area_mixin.py

```diff
@@ -252,6 +252,21 @@ def get_student_submission_context(self, student_username, submission):
         Returns:
             A context dict for rendering a student submission and associated rubric (for staff grading).
         """
+        def should_use_user_state():
+            """
+            Return a boolean if the user state is used for additional data checks.
+
+            User state is utilized when all of the following are true:
+            1. user state service is available(which is only part of courseware)
+            2. The waffle flag/switch is enabled
+            3. the file upload is required or optional
+            4. the file data from submission is missing information
+            """
+            return self.is_user_state_service_available() \
+                and self.user_state_upload_data_enabled() \
+                and self.file_upload_response \
+                and not any(context['staff_file_urls'])
```

Comment:
Nit: I know this works because `context` is already defined in the local scope in which this function is executed, but it would be clearer if this helper function simply took `staff_file_urls` as a parameter, so that you would call this function as `if should_use_user_state(context['staff_file_urls']):`

Reply:
Also, I'd recommend making the `not any(staff_file_urls)` the first thing we check, since we don't have to check anything else if that list contains at least one file URL.

Reply:
Ok. I will address both comments.